Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -317,26 +321,38 @@ export async function getProvisioningProfileType(provProfilePath: string) {
             throw tl.loc('ProvProfileDetailsNotFound', provProfilePath);
         }
 
-        //get ProvisionsAllDevices - this will exist for enterprise profiles
+        // get ProvisionsAllDevices - this will exist for developer id/enterprise profiles
         let provisionsAllDevices: string = await printFromPlist('ProvisionsAllDevices', tmpPlist);
         tl.debug('provisionsAllDevices = ' + provisionsAllDevices);
         if (provisionsAllDevices && provisionsAllDevices.trim().toLowerCase() === 'true') {
-            //ProvisionsAllDevices = true in enterprise profiles
-            provProfileType = 'enterprise';
+            // Bypass the app store, distribute a signed app directly to end users
+            if (macOSProfile) {
+                provProfileType = 'developer-id';
+            }
+            else {
+                provProfileType = 'enterprise';
+            }
         } else {
             let getTaskAllow: string = await printFromPlist('Entitlements:get-task-allow', tmpPlist);
             tl.debug('getTaskAllow = ' + getTaskAllow);
             if (getTaskAllow && getTaskAllow.trim().toLowerCase() === 'true') {
-                //get-task-allow = true means it is a development profile
+                // get-task-allow = true means it is an iOS development profile
                 provProfileType = 'development';
             } else {
                 let provisionedDevices: string = await printFromPlist('ProvisionedDevices', tmpPlist);
                 if (!provisionedDevices) {
                     // no provisioned devices for non-development profile means it is an app-store profile
                     provProfileType = 'app-store';
                 } else {
```

Comment:
Should this be `} else if (isMacOSProfile) {` (collapsing the if-block below into this one)?  Seems like this whole if-block is for the purpose of setting `provProfileType`.